Replicator/Worker.cc

```diff
@@ -176,6 +176,9 @@ namespace litecore { namespace repl {
         if (err.domain == "HTTP"_sl) {
             domain = WebSocketDomain;
             code = err.code;
+        } else if (err.domain == "BLIP"_sl && err.code == 404) {
+            domain = LiteCoreDomain;
```

Comment:
Did we way that just let the code flow to the end and (having the same effect) ?

Reply:
It has almost the same effect, apart from it also logs "Received unknown error from remote", which is A. One extra unnecessary logging call and B. The error is known, so slightly inaccurate.

Reply:
I meant you may remove this "else if" and won't change anything, right? Without this "else if", it will progress to the following "else', and, if in the future, we decide to add a custom error for it, it will be picked up automatically.
By the way, I don't see "log" in this "else if".

Reply:
It will hit the log in the final `else` if this elseif does not exist, which I wanted to avoid because of the extra log. But no, it won't change any functionality.

Reply:
I see you wanted avoid that extra log, but you may change the log there. 
if (code == 0) { if (err.domain == "BLIP") ...
This makes the code clear that there is no domain in Core named "BLIP"

Reply:
The impression of this "else if" is that you wanted to avoid the following "else", but I think if we have general treatment of BLIP error, we should let the general error logic to work.

Reply:
Yeah, I understand your meaning now. I do agree